### PR TITLE
Tune ansible.cfg

### PR DESCRIPTION
**Pull request by jtudelag**
Based on the OpenShift one:
https://docs.openshift.com/container-platform/3.7/scaling_performance/install_practices.html#scaling-performance-install-optimization

* Increases number of forks.
* Disables host_key_checking
* Smart gathering facts
* Fact caching jsonfile
* Enables profile_tasks callback
* Mutliplexes ssh connections (ControlMaster)
* Enables pipelining

### CODE REVIEW

File: ansible.cfg [hunk 1]
```diff
@@ -8,6 +8,15 @@ roles_path = ./roles
...
 log_path = /var/log/ansible.log
 
+forks = 20 
+host_key_checking = False
+gathering = smart
+fact_caching = jsonfile
+fact_caching_connection = $HOME/ansible/facts
+fact_caching_timeout = 600
+nocows = 1
+callback_whitelist = profile_tasks
```
Comment (leseb) [inline ansible.cfg]: this is being called by tox already, but it's important to have it here too

File: ansible.cfg [hunk 2]
```diff
@@ -8,6 +8,15 @@ roles_path = ./roles
 # Be sure the user running Ansible has permissions on the logfile
 log_path = /var/log/ansible.log
 
+forks = 20 
+host_key_checking = False
+gathering = smart
+fact_caching = jsonfile
+fact_caching_connection = $HOME/ansible/facts
```
Comment (leseb) [inline ansible.cfg]: Will ansible create this directory?

**Comment by leseb** [comment]
jenkins test luminous-ansible2.4-purge_bluestore_osds_container

File: ansible.cfg (hunk 2 above)
Comment (jtudelag) [reply ansible.cfg]: Yes, Ansible creates it the first time it gathers facts.

**Pull request merged**